CuratedContent/RequirementsAndRequirementsEngineering.md

```diff
@@ -0,0 +1,38 @@
+# Overview of Requirements and Requirements Engineering
+
+A software **requirement** is a description of a specific capability that a software product is expected to have in order to satisfy the needs of stakeholders. While there is no single, agreed-upon definition of the term, it encompasses both the things that the software must do and the conditions under which it must do them. **Requirements engineering** is the process of formally identifying, documenting, and validating software requirements. For software engineers, requirements engineering represents the first steps in the development of a software product; requirements translate into specifications which then inform the design and implementation. 
+
+Special emphasis is given to the requirements process as it is well-known that the most painful struggles and the most spectacular failures in software development stem from oversights and misestimations very early on in the life of a project (Reel 1999). Scientific software is no exception. Studies conducted by NASA in the early 1990s found that while cost overruns were rampant among R&D projects, projects which invested 2% to 3% of their budget on planning and developing requirements saw cost overruns between 80-200%, and those that invested 8% to 14% of their budget had overruns between 0-50% (Hihn and Habib-Agahi 1991, Habib-Agahi et al. 1991). The benefits are well-established: requirements engineering, by far the least expensive development activity, has an outsized impact on everything that follows.
+
+Despite this, numerous studies have found that scientific software developers, as a rule, do not produce requirements documents (Segal 2005, Segal 2009, Sanders and Kelly 2008, Li et al. 2011, Heaton and Carver 2015). There are several reasons for this, perhaps the greatest one being that researchers simply don't know how. Many scientific software projects are exploratory in character, and the requirements are discovered in the course of development. Trying to produce complete, well-articulated requirements upfront would be an exercise in futility. However, with some considerations and adaptations, readers will find that requirements engineering techniques can yield tremendous benefits.
```

Comment:
I think this is a very important point: Most scientific software nowadays is written by scientists that are experts in their specific domain, but aren't programmers or software engineers. Although this is changing due to software engineering becoming more popular / more well known,  this is a major obstacle for many project imho.
Maybe you can comment on this issue in a little more depth.

Reply:
Nice start, Reed.  Info about software requirements that speaks to the science community will be a great resource.